lib/es6-promise/polyfill.js

```diff
@@ -2,7 +2,7 @@
 import Promise from './promise';
 
 export default function polyfill() {
-  var local = (typeof global !== 'undefined') ? global : self;
+  var local = Function('return this;')();
```

Comment:
i believe some of these environments disallow the statement all together. Is their no way to just use `this` without the eval in that context?